Fix nodejs application issue with mongodb

jcpowermac opened the pull request:
- Since the template is being process and applied with each
  job run the secret is also being updated since it contains generate
  values from the template.  The mongodb DC is not again rolled out
  and the secret is not updated.  To resolve this the secret is
  delete from the processed template.

codificat commented on Jenkinsfile:
```diff
@@ -48,17 +48,35 @@ pipeline {
+                             * the case of using a database along with an application.
+                             * Issue: Since the template is being process and applied with each
+                             * job run the secret is also being updated since it contains generate
+                             * values from the template.
```
This makes me wonder if we should take this one step further: do we need to process/apply the whole template on every run? The update to the other objects in the template might not be as harmful as the secret's, but isn't it unnecessary anyway?

codificat reviewed (commented):
The change looks good to me as a fix for this particular issue with the secret in the sample application/template, but wondering if we shouldn't take a different approach instead and reconsider the whole template processing on each pipeline run as a more generic solution?

jcpowermac commented:
@codificat ok take a look at the changes when you get a chance.

jcpowermac commented:
Found a bug during testing with Jenkins 2.73.2...back to WIP.

jcpowermac committed:
Fix nodejs application issue with mongodb

- Since the template is being process and applied with each
  job run the secret is also being updated since it contains generate
  values from the template.  The mongodb DC is not again rolled out
  and the secret is not updated.  To resolve this the secret is
  delete from the processed template.
- Added the ability to check if the checkout has a new version of the template.

codificat reviewed (approved):
Great work! LGTM
